actors/builtin/miner/miner_test.go

```diff
@@ -2616,6 +2679,253 @@ func TestChangeMultiAddrs(t *testing.T) {
+		secondExpiation := periodOffset + 20
+		actor.changeBeneficiary(rt, actor.owner, beneficiaryChange{secondBeneficiaryId, secondQuota, secondExpiation}, nil)
+		info := actor.getInfo(rt)
+		require.EqualValues(t, secondBeneficiaryId, info.PendingBeneficiaryTerm.NewBeneficiary)
+		require.EqualValues(t, secondQuota, info.PendingBeneficiaryTerm.NewQuota)
+		require.EqualValues(t, secondExpiation, info.PendingBeneficiaryTerm.NewExpiration)
+
+		//confirm by old beneficiary address
+		actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{secondBeneficiaryId, secondQuota, secondExpiation}, nil)
+		info = actor.getInfo(rt)
+		require.EqualValues(t, secondBeneficiaryId, info.PendingBeneficiaryTerm.NewBeneficiary)
+		require.EqualValues(t, secondQuota, info.PendingBeneficiaryTerm.NewQuota)
+		require.EqualValues(t, secondExpiation, info.PendingBeneficiaryTerm.NewExpiration)
+
+		//confirm by new beneficiary address and worked
+		actor.changeBeneficiary(rt, secondBeneficiary, beneficiaryChange{secondBeneficiaryId, secondQuota, secondExpiation}, &secondBeneficiaryId)
+		info = actor.getInfo(rt)
+		require.EqualValues(t, secondBeneficiaryId, info.Beneficiary)
+		require.EqualValues(t, secondQuota, info.BeneficiaryTerm.Quota)
+		require.EqualValues(t, secondExpiation, info.BeneficiaryTerm.Expiration)
+		require.EqualValues(t, big.Zero(), info.BeneficiaryTerm.UsedQuota)
+		require.EqualValues(t, (*miner.PendingBeneficiaryChange)(nil), info.PendingBeneficiaryTerm)
+	})
+
+	t.Run("failure approval has different params", func(t *testing.T) {
+		rt, actor := setupFunc()
+		actor.constructAndVerify(rt)
+
+		quota := abi.NewTokenAmount(100)
+		expiration := periodOffset * 2
+
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, quota, expiration + 1}, &firstBeneficiary)
+		})
+
+		actor.changeBeneficiary(rt, actor.owner, beneficiaryChange{firstBeneficiaryId, quota, expiration}, nil)
+		// assert change has been made in state
+		info := actor.getInfo(rt)
+		require.EqualValues(t, expiration, info.PendingBeneficiaryTerm.NewExpiration)
+		require.EqualValues(t, quota, info.PendingBeneficiaryTerm.NewQuota)
+		require.EqualValues(t, firstBeneficiaryId, info.PendingBeneficiaryTerm.NewBeneficiary)
+
+		rt.SetEpoch(periodOffset + 10)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, quota, expiration + 1}, &firstBeneficiary)
+		})
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, big.Add(quota, big.NewInt(1)), expiration}, &firstBeneficiary)
+		})
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{secondBeneficiaryId, big.Add(quota, big.NewInt(1)), expiration}, &firstBeneficiary)
+		})
```

Comment:
For completeness, show a successful call with the correct parameters here, demonstrating that it would have succeeded. 

(If you want, you can collapse this with other tests, e.g. by moving these failure cases up to "successfully change beneficiary to another address". Your call, both ways are good for me).